integrated_channels/moodle/client.py

```diff
@@ -311,38 +312,200 @@ def _wrapped_create_course_completion(self, user_id, payload):
+
+    def get_announcement_post(self, forum_id):
+        """
+        Returns the id of the discussion post matching our announcement or returns nothing.
+        """
+        response = self._get_forum_discussions(forum_id)
+        response_text = json.loads(response.text)
+        for _, discussion in enumerate(response_text['discussions']):
+            if discussion['subject'] == ANNOUNCEMENT_POST_SUBJECT:
+                return discussion['id']
+        return ''
+
+    def _update_announcement_post(self, course_id, announcement):
+        """
+        Updates announcement post with announcement text composed in the exporter.
+        """
+        forum_id = self.get_announcement_forum(course_id)
+        post_id = self.get_announcement_post(forum_id)
+        params = {
+            'wsfunction': 'mod_forum_update_discussion_post',
+            'postid': post_id,
+            'message': announcement,
+        }
+        return self._post(params)
+
     @moodle_request_wrapper
+    def _create_forum_post(self, course_id, announcement):
+        """
+        Creates a new Announcement post featuring edX course details and link.
+        """
+        forum_id = self.get_announcement_forum(course_id)
+        params = {
+            'wsfunction': 'mod_forum_add_discussion',
+            'forumid': forum_id,
+            'subject': ANNOUNCEMENT_POST_SUBJECT,
+            'options[0][name]': 'discussionpinned',
+            'options[0][value]': 'true',
+            'message': announcement
+        }
+        return self._post(params)
+
+    @moodle_request_wrapper
+    def _delete_content_metadata(self, course_id):
+        """
+        unified deletion parameters and call.
+        For main delete content metadata task and for cleanup.
+        """
+        params = {
+            'wsfunction': 'core_course_delete_courses',
+            'courseids[]': course_id
+        }
+        return self._post(params)
+
+    def create_content_metadata(self, serialized_data):
+        """
+        This content metadata creation has 3 phases:
+        1. Create the course in Moodle with most information
+        2. Create announcement post with course summary and details
+        3. Update the course format to "singleactivity" to show course details.
+        """
+        serialized_data['wsfunction'] = 'core_course_create_courses'
+        announcement = serialized_data.pop('courses[0][announcement]')
+        course_format = serialized_data.pop('courses[0][format]')
+        response = self._wrapped_create_content_metadata(serialized_data)
+        # Response format should be [{"id":1, "shortname": "name"},{...}]
+        if response[0].get('id', None):
+            try:
+                post = self._create_forum_post(
+                    response[0].get('id'),
+                    announcement
+                )
+            except (TypeError, IndexError):
+                self._delete_content_metadata(response[0].get('id'))
```

Comment:
I'm just calling this and hoping for the best, because if it fails to delete a course we're in a really bad state that we can't actually fix. :(